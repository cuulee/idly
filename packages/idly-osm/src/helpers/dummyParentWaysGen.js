"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var immutable_1 = require("immutable");
function dummyParentWaysGen(obj) {
    return immutable_1.Map(obj);
}
exports.dummyParentWaysGen = dummyParentWaysGen;
