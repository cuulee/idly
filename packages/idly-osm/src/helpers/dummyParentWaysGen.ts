import { ParentWays } from 'idly-common/lib/osm/structures';

export function dummyParentWaysGen(obj: any) {
  const parentWays: ParentWays = new Map();
  Object.keys(obj).forEach(k => parentWays.set(k, obj[k]));
  return parentWays;
}
