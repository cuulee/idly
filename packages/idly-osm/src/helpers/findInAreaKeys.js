"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var areaKeys_1 = require("idly-data/lib/areaKeys/areaKeys");
// `highway` and `railway` are typically linear features, but there
// are a few exceptions that should be treated as areas, even in the
// absence of a proper `area=yes` or `areaKeys` tag.. see #4194
var lineKeys = {
    highway: {
        rest_area: true,
        services: true
    },
    railway: {
        roundhouse: true,
        station: true,
        traverser: true,
        turntable: true,
        wash: true
    }
};
function findInAreaKeys(tags) {
    var areaKeys = areaKeys_1.getAreaKeys();
    for (var key in tags) {
        if (key in areaKeys && !(tags[key] in areaKeys[key])) {
            return true;
        }
        if (key in lineKeys && tags[key] in lineKeys[key]) {
            return true;
        }
    }
    return false;
}
exports.findInAreaKeys = findInAreaKeys;
