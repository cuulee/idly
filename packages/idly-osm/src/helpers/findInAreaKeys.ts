import { Tags } from 'idly-common/lib';

import { areaKeys } from '../presets/presets';

export const findInAreaKeys = (tags: Tags) => {
  let found = false;
  tags.forEach((v, key) => {
    if (areaKeys.has(key) && !areaKeys.hasIn([key, v])) {
      found = true;
      return false;
    }
  });
  return found;
};
