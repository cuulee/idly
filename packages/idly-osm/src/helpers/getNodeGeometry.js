"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var structures_1 = require("idly-common/lib/osm/structures");
function getNodeGeometry(id, parentWay) {
    if (parentWay.size === 0)
        return structures_1.OsmGeometry.POINT;
    return structures_1.OsmGeometry.VERTEX;
}
exports.getNodeGeometry = getNodeGeometry;
