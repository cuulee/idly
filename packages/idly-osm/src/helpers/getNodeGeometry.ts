import {
  EntityId,
  NodeGeometry,
  OsmGeometry,
  ParentWays
} from 'idly-common/lib/osm/structures';

export function getNodeGeometry(
  id: EntityId,
  parentWay: Set<string>
): NodeGeometry {
  if (parentWay.size === 0) return OsmGeometry.POINT;

  return parentWay.size === 1 ? OsmGeometry.VERTEX : OsmGeometry.VERTEX_SHARED;
}
