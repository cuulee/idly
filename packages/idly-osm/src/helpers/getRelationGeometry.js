"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var structures_1 = require("idly-common/lib/osm/structures");
// @TOFIX this stub
function getRelationGeometry() {
    return structures_1.OsmGeometry.RELATION;
}
exports.getRelationGeometry = getRelationGeometry;
