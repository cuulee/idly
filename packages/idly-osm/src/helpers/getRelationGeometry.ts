import { OsmGeometry } from 'idly-common/lib';

// @TOFIX this stub
export function getRelationGeometry() {
  return OsmGeometry.RELATION;
}
