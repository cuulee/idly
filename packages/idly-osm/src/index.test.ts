import {
  entityTableGen,
  genLngLat,
  nodeFactory,
  tagsFactory,
  wayFactory
} from 'idly-common/lib';
import { dummyParentWaysGen } from './helpers/dummyParentWaysGen';

import { IdlyOSM } from './index';

describe('test IdlyOSM', () => {
  describe('onParseEntities', () => {
    const n1 = nodeFactory({
      id: 'n1',
      loc: genLngLat({ lon: 15, lat: 10 }),
      tags: tagsFactory({ k: 'k' })
    });
    const n2 = nodeFactory({
      id: 'n2',
      loc: genLngLat({ lon: 5, lat: 12 }),
      tags: tagsFactory({ k: 'k' })
    });
    const n3 = nodeFactory({
      id: 'n3',
      loc: genLngLat({ lon: 15, lat: 10 }),
      tags: tagsFactory({ k: 'k' })
    });
    const n4 = nodeFactory({
      id: 'n4',
      loc: genLngLat({ lon: 0, lat: 10 }),
      tags: tagsFactory({ k: 'k' })
    });

    const w1 = wayFactory({
      id: 'w1',
      nodes: ['n1', 'n2', 'n4'],
      tags: tagsFactory({ highway: 'residential' })
    });

    const w2 = wayFactory({
      id: 'w2',
      nodes: ['n1', 'n3'],
      tags: tagsFactory({ highway: 'residential' })
    });
    var table = entityTableGen(new Map(), [n1, n2, n3, n4, w1, w2]);
    var parentWays = dummyParentWaysGen({
      n1: new Set(['w1', 'w2']),
      n2: new Set(['w1']),
      n3: new Set(['w2']),
      n4: new Set(['w1'])
    });
    it('should behave...', () => {
      var mainClass = new IdlyOSM();
      var propsAdded = mainClass.onParseEntities(table, parentWays);
      expect(propsAdded).toMatchSnapshot();
    });
  });
});
