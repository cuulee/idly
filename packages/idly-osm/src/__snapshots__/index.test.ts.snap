// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test IdlyOSM onParseEntities should behave... 1`] = `
Map {
  "n1" => Object {
    "geometry": "vertex_shared",
    "icon": "circle",
    "name": undefined,
  },
  "n2" => Object {
    "geometry": "vertex",
    "icon": "circle",
    "name": undefined,
  },
  "n3" => Object {
    "geometry": "vertex",
    "icon": "circle",
    "name": undefined,
  },
  "n4" => Object {
    "geometry": "vertex",
    "icon": "circle",
    "name": undefined,
  },
  "w1" => Object {
    "geometry": "line",
    "icon": "highway-residential",
    "name": undefined,
    "tagsClass": "tag-highway",
    "tagsClassType": "tag-highway-residential",
  },
  "w2" => Object {
    "geometry": "line",
    "icon": "highway-residential",
    "name": undefined,
    "tagsClass": "tag-highway",
    "tagsClassType": "tag-highway-residential",
  },
}
`;
